athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/handlers/MetadataHandler.java

```diff
@@ -446,6 +452,21 @@ public void onPing(PingRequest request)
         //NoOp
     }
 
+    /**
+     * Helper function used to check if the account that calls the lambda function owns the spill bucket
+     *
+     * @param s3 The S3 object for the account.
+     * @param bucket The name of the spill bucket.
+     */
+    @VisibleForTesting
+    void checkBucketAuthZ(final AmazonS3 s3, final String bucket)
+    {
+        Set<String> buckets = s3.listBuckets().stream().map(b -> b.getName()).collect(Collectors.toSet());
```

Comment:
IsI dont think we need to do this more than once so we should cache the check and only re-check if the bucket name changes.

Reply:
This is easily fixable. I'll make the changes once we settle on where the checking should be.

Reply:
Regardless of where we do the check, caching makes sense.